openc3-cmd-tlm-api/app/controllers/gems_controller.rb

```diff
@@ -34,34 +34,36 @@ def create
       begin
         gem_file_path = temp_dir + '/' + file.original_filename
         FileUtils.cp(file.tempfile.path, gem_file_path)
-        result = OpenC3::GemModel.put(gem_file_path, scope: params[:scope])
+        OpenC3::GemModel.put(gem_file_path, gem_install: true, scope: params[:scope])
         OpenC3::Logger.info("Gem created: #{params[:gem]}", scope: params[:scope], user: user_info(request.headers['HTTP_AUTHORIZATION']))
+        head :ok
```

Comment:
I think head :ok is supposed to be the last line in the controller because it returns the 200 code. Not sure how this works with the ensure below.

Reply:
It works because only on render or head gets run.

Reply:
The ensure happens in both cases, but rails only wants you to render once.   One of the tricky things in rails is that you can render in the middle of a function and then other stuff after.   renders do not return from the function.